### PR TITLE
[Feature] Support save last checkpoint

### PR DESCRIPTION
**Pull request by xvjiarui**

### CODE REVIEW

File: mmcv/runner/hooks/checkpoint.py [hunk 1]
```diff
@@ -35,6 +36,7 @@ def __init__(self,
                  out_dir=None,
                  max_keep_ckpts=-1,
                  sync_buffer=False,
+                 save_last=True,
```
Comment (hellock) [inline mmcv/runner/hooks/checkpoint.py:39]: Better to put `save_last` together with `max_keep_ckpts`.

File: mmcv/runner/hooks/checkpoint.py [hunk 2]
```diff
@@ -26,6 +26,7 @@ class CheckpointHook(Hook):
             Default: -1, which means unlimited.
         sync_buffer (bool): Whether to synchronize buffers in different
             gpus. Default: False.
+        keep_last (bool): Whether to save last checkpoint.
```
Comment (hellock) [inline mmcv/runner/hooks/checkpoint.py:29]: save_last (bool): Whether to force the last checkpoint to be saved regardless of interval.

File: mmcv/runner/hooks/checkpoint.py [hunk 3]
```diff
@@ -99,3 +102,10 @@ def after_train_iter(self, runner):
         if self.sync_buffer:
             allreduce_params(runner.model.buffers())
         self._save_checkpoint(runner)
+
+    def after_run(self, runner):
+        if self.save_last:
+            if not self.out_dir:
+                self.out_dir = runner.work_dir
```
Comment (hellock) [inline mmcv/runner/hooks/checkpoint.py:109]: Maybe this two lines can be moved to the init method?

Comment (xvjiarui) [reply mmcv/runner/hooks/checkpoint.py]: But the checkpoint hook can't access runner during `__init__`. Moved to `before_run` instead.

**Review by hellock — APPROVED** [review]

**Comment by hellock** [comment]
@xvjiarui Unit tests failed. There might be bugs.

**Comment by xvjiarui** [comment]
@hellock Note that I set the `save_last` default to be true, so the original behavior is changed.

**Comment by hellock** [comment]
> @hellock Note that I set the `save_last` default to be true, so the original behavior is changed.

This should be okay.

**Pull request merged**